### PR TITLE
stm32l0: Timer support

### PR DESCRIPTION
**Pull request by rkreis**
Based on #571

### CODE REVIEW

**Comment by fenugrec** [comment]
Double-checked the defines vs RM0367, RM0376, RM0377, everything looks oK

File: include/libopencm3/stm32/l0/rcc.h [hunk 1]
```diff
@@ -291,8 +291,10 @@
 #define RCC_APB1RSTR_SPI2RST			(1 << 14)
 #define RCC_APB1RSTR_WWDGRST			(1 << 11)
 #define RCC_APB1RSTR_LCDRST			(1 << 9)
-#define RCC_APB1RSTR_TIM6RST			(1 << 4)
 #define RCC_APB1RSTR_TIM2RST			(1 << 0)
+#define RCC_APB1RSTR_TIM3RST			(1 << 1)
+#define RCC_APB1RSTR_TIM6RST			(1 << 4)
+#define RCC_APB1RSTR_TIM7RST			(1 << 5)
```
Comment (karlp) [inline include/libopencm3/stm32/l0/rcc.h]: you've switched the ordering here.

File: include/libopencm3/stm32/l0/rcc.h [hunk 2]
```diff
@@ -350,8 +352,10 @@
 #define RCC_APB1ENR_SPI2EN			(1 << 14)
 #define RCC_APB1ENR_WWDGEN			(1 << 11)
 #define RCC_APB1ENR_LCDEN			(1 << 9)
-#define RCC_APB1ENR_TIM6EN			(1 << 4)
 #define RCC_APB1ENR_TIM2EN			(1 << 0)
+#define RCC_APB1ENR_TIM3EN			(1 << 1)
+#define RCC_APB1ENR_TIM6EN			(1 << 4)
+#define RCC_APB1ENR_TIM7EN			(1 << 5)
```
Comment (karlp) [inline include/libopencm3/stm32/l0/rcc.h]: likewise

File: lib/stm32/l0/timer.c [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <libopencm3/stm32/timer.h>
```
Comment (karlp) [inline lib/stm32/l0/timer.c]: Why do you need to create this file?

**Comment by rkreis** [comment]
Fixed the issues addressed by @karlp.

**Comment by karlp** [comment]
merged via rebase with updates in 9b642de

**Pull request closed without merging**